src/test/java/org/apache/maven/plugins/shade/mojo/ShadeMojoTest.java

```diff
@@ -60,6 +64,50 @@
 public class ShadeMojoTest
     extends PlexusTestCase
 {
+    public void testDefaultConfiguration() throws Exception
```

Comment:
Very creative with reflection, but I wonder if we are testing the right things. If it is too hard to verify it with a unittest, why not switch to an integration test?

Reply:
Thought about it but I found it saner as a compromise cause this is really more a whitebox test to check we mutate the user config as expected than an IT which would be more complicated with fewer guarantee on what we do at that level IMHO (here we really want to check the default config is setup and not smoke test it, at least was the intention)